### PR TITLE
[ENHANCEMENT] Easy delete data

### PR DESCRIPTION
**Pull request by gantunesr**
**Description**

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_
_1. What is the reason for the change?_

Update "Easy Delete Data" feature to be compatible with Segment

_2. What is the improvement/solution?_

Use PR https://github.com/MetaMask/metamask-mobile/pull/4389 as reference.

**Screenshots/Recordings**

Use PR https://github.com/MetaMask/metamask-mobile/pull/4389 as reference.

**Issue**

Progresses https://github.com/MetaMask/mobile-planning/issues/373

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

### CODE REVIEW

File: app/core/Analytics/MetaMetrics.ts [hunk 1]
```diff
@@ -249,6 +250,12 @@ class MetaMetrics implements IMetaMetrics {
     }
   };
 
+  #getDeleteRegulationId = async (): Promise<string> =>
```
Comment (owencraston) [inline app/core/Analytics/MetaMetrics.ts:253]: I am not sure what you are using the `#` but if you are using it to mark the method as private, you can simply use  [private class member in typescript ](https://www.typescriptlang.org/docs/handbook/2/classes.html)

File: app/components/Views/Settings/SecuritySettings/Sections/DeleteMetaMetricsData.tsx [hunk 2]
```diff
@@ -102,28 +102,9 @@ const DeleteMetaMetricsData = () => {
     }
   };
 
-  const checkDataDeleteStatus = useCallback(async () => {
```
Comment (owencraston) [inline app/components/Views/Settings/SecuritySettings/Sections/DeleteMetaMetricsData.tsx:105]: does this feature assume that we have migrated off of mixpanel already ?

**Review by owencraston — APPROVED** [review]

**Review by tommasini — COMMENTED** [review]
Awesome! Left a question

File: app/components/Views/Settings/SecuritySettings/Sections/DeleteMetaMetricsData/DeleteMetaMetricsData.tsx [hunk 3]
```diff
@@ -101,28 +80,9 @@ const DeleteMetaMetricsData = () => {
...
-        await checkDataDeleteStatus();
-        setDeletionTaskDate(Analytics.getDeletionTaskDate());
-      }
-    };
-
-    setHasCollectedData(Analytics.getIsDataRecorded() || enableDeleteData());
-
-    checkStatus();
-  }, [checkDataDeleteStatus, enableDeleteData, dataDeleteStatus]);
+    setHasCollectedData(MetaMetrics.getIsDataRecorded() && enableDeleteData());
```
[the remark this answers is not in the document]
Comment (gantunesr) [reply app/components/Views/Settings/SecuritySettings/Sections/DeleteMetaMetricsData/DeleteMetaMetricsData.tsx:84]: It's not only executed on the first render, this value is reset every time `enableDeleteData` or `dataDeleteStatus` changes

[the remark this answers is not in the document]
Comment (gantunesr) [reply app/components/Views/Settings/SecuritySettings/Sections/DeleteMetaMetricsData/DeleteMetaMetricsData.tsx:84]: Lets move this conversation to this PR https://github.com/MetaMask/metamask-mobile/pull/5139

**Pull request merged**